File: src/Python/Parcial2/ActividadP2A4/trabajadorpiezas.py
from empleado import Empleado

class TrabajadorPiezas(Empleado):
    def __init__(self, nombre, apellido, nss, salario, piezas):
        super().__init__(nombre, apellido, nss)
        self.establecerSalario(salario)
        self.establecerPiezas(piezas)

    def establecerSalario(self, salario):
        if float(salario) < 0.0:
            self.salario = 0.0
        else:
            self.salario = float(salario)

    def obtenerSalario(self):
        return self.salario

    def establecerPiezas(self, piezas):
        if int(piezas) < 0:
            self.piezas = 0
        else:
            self.piezas = int(piezas)

    def obtenerPiezas(self):
        return self.piezas

    def ingresos(self):
        return self.obtenerSalario() * self.obtenerPiezas()

    def __str__(self):
        return f"trabajador por piezas: {super().__str__()}\nsalario por pieza: ${self.obtenerSalario()}; piezas producidas: {self.obtenerPiezas()}"

File: src/C++/Parcial2/ActividadP2A4/Origen/TrabajadorPiezas.cpp
#include "TrabajadorPiezas.h"
#include <sstream>
#include <iomanip>

TrabajadorPiezas::TrabajadorPiezas(const std::string& nombre, const std::string& apellido, const std::string& nss, double salarioBase, int piezasHechas)
    : Empleado(nombre, apellido, nss) {
    establecerSalario(salarioBase);
    establecerPiezas(piezasHechas);
}

void TrabajadorPiezas::establecerSalario(double salarioBase) {
    salario = (salarioBase < 0.0) ? 0.0 : salarioBase;
}

double TrabajadorPiezas::obtenerSalario() const {
    return salario;
}

void TrabajadorPiezas::establecerPiezas(int piezasHechas) {
    piezas = (piezasHechas < 0) ? 0 : piezasHechas;
}

int TrabajadorPiezas::obtenerPiezas() const {
    return piezas;
}

double TrabajadorPiezas::ingresos() const {
    return obtenerSalario() * obtenerPiezas();
}

std::string TrabajadorPiezas::toString() const {
    std::ostringstream output;
    output << "trabajador por piezas: " << Empleado::toString() 
           << "\nsalario por pieza: $" << std::fixed << std::setprecision(2) << obtenerSalario()
           << "; piezas producidas: " << obtenerPiezas();
    return output.str();
}
